### ISSUE TITLE
Problem with getOccurrencesAfter

### ISSUE THREAD

**Issue by justageek** [rlanvin/php-rrule#135]
I am debugging an issue with this method use in our code, and I have found that RRule indeed finds 2 occurences after a date, that look like

```
Array
(
    [0] => DateTime Object
        (
            [date] => 2023-08-28 04:05:33.000000
            [timezone_type] => 1
            [timezone] => +00:00
        )

    [1] => DateTime Object
        (
            [date] => 2023-09-28 04:05:33.000000
            [timezone_type] => 1
            [timezone] => +00:00
        )

)
```

But this this code in the library 

```
return array_slice($occurrences, 1);
```

effectively skips the first item in the array, why does it do this?  I want the 2023-08-28 04:05:33.000000 as it is the correct next date.

**Comment by rlanvin** [comment]
Can you provide a code snippet that demonstrates the issue and that I can use to reproduce it?

**Comment by justageek** [comment]
```php
    $rule = new RRule('RRULE:FREQ=MONTHLY;INTERVAL=1;BYMONTHDAY=28');
    $next = $rule->getOccurrencesAfter(Carbon::parse('2023-07-28 04:00:33.433843+0000'), false, 1);
    dd($next);
```

Note this code would have been executed at the date time in the example code, so that affects how the RRule constructor established DTSTART I believe.  This is from Laravel thus the use of dd() in a test route to see the output.  So, i had to hack in this in your class to simulate what happened on the server:
```php
$this->dtstart = new \DateTime('2023-07-28 04:00:33.433843+0000'); 
```

**Comment by justageek** [comment]
in reality the server code does something like this:
```php
$lastRun = Carbon::now(new DateTimeZone('UTC'));
$nextOccurrences = $rrule->getOccurrencesAfter($lastRun, false, 1);
```

An when this ran on July 28 at 4AM UTC, the next occurence came out as 
```php
2023-09-28 04:00:00
```

**Comment by rlanvin** [comment]
I tried to reproduce your problem with the following code:

```php
$now = new DateTime('2023-07-28 04:00:33.433843+0000');
$rule = new RRule('RRULE:FREQ=MONTHLY;INTERVAL=1;BYMONTHDAY=28', $now);
$next = $rule->getOccurrencesAfter($now, false, 1);
print_r($next);
```

And I get the expected result (August):

```
Array
(
    [0] => DateTime Object
        (
            [date] => 2023-08-28 04:00:33.000000
            [timezone_type] => 1
            [timezone] => +00:00
        )

)
```

The reason `getOccurrencesAfter` slices the array when you set you set `$inclusive` to `false` (which is the default behaviour) **and when `$start` is a valid occurrence**, is because it's just a helper method that calls `getOccurrencesBetween($start, null, $limit)`. 

`getOccurrencesBetween` will always return the bounds if they are also valid occurrences (i.e. it's always inclusive). So if you'd pass a known occurrence to `getOccurrencesAfter` and set `$inclusive` to `false`, and if the array wasn't sliced, then you would always just get the occurrence you passed as a result, which wouldn't be very helpful...

You see [here](https://github.com/rlanvin/php-rrule/blob/master/src/RRuleTrait.php#L104) that if you set `$inclusive` to `true` **or** if `$date` is not an occurrence, then it will not slice anything.

If you say `getOccurrencesBetween` returned August and September, and August got sliced out, and `$inclusive` is set to `false` that can only means that `occursAt($date)` returned `true`, which means only two you passed the August occurrence to `getOccurrencesAfter`...

So unless you can provide a code to reproduce the issue in isolation (and that doesn't rely on modifying the library internals), I'm afraid it looks like the problem might be elsewhere in your own code.

I can suggest to make sure you use the latest version of the library just in case.

**Comment by justageek** [comment]
Ok, thanks much, let me test your code and inspect mine as well.

**Comment by justageek** [comment]
Disregard my last comment, I had not properly upgraded to 2.4.1, that appears to fix our problem.

**Comment by rlanvin** [comment]
Ok glad to hear the latest version works, but that's interesting - I wonder what kind of issue could have caused that behaviour and I don't remember any similar bug. What version were you using before upgrading?

Also feel free to close this ticket if your problem is resolved.

**Comment by justageek** [comment]
We were using 2.3.0 before, and it baffles me too  as to why it would be different.

**Comment by rlanvin** [comment]
Ah nice! Then I think you might have encountered this bug that was fixed in 2.3.1: https://github.com/rlanvin/php-rrule/issues/104
